### ISSUE TITLE
bug: lowercase for in htmlFor attribute doesn't work

### ISSUE THREAD

**Issue by katiedeangelis** [ionic-team/stencil#3473]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.17.1

### Current Behavior

In my render function, I created a checkbox:

```
<label htmlFor={this.id}>My Label Text</label>
<input type="checkbox" id={this.id}></input>
```

When the component renders, "htmlFor" becomes "htmlfor" and the "My Label Text" does not toggle the checkbox.

### Expected Behavior

In my render function, when "htmlFor" is added to a checkbox, that is the attribute (with case) that should render. When the component renders, I also expect that when the "My Label Text" label is clicked, the checkbox is toggled.

### Steps to Reproduce

1. Create a checkbox in a render function with the following label/input setup:

```
<label htmlFor={this.id}>My Label Text</label>
<input type="checkbox" id={this.id}></input>
```
2. When the component renders, click the label
3. Notice the checkbox does not toggle

### Code Reproduction URL

#

### Additional Information

I noticed in the stencil-public-runtime.ts file that there is an htmlfor element property on the LabelHTMLAttributes interface which is not correct:

```
   interface LabelHTMLAttributes<T> extends HTMLAttributes<T> {
        form?: string;
        htmlFor?: string;
        htmlfor?: string;
    }
```

**Comment by alicewriteswrongs** [comment]
Hi @katiedeangelis, thanks for filing this issue and for filing the related PR! I get the gist of what you're reporting here, but would you mind creating a simple reproduction repo that showcases this issue? Doing so will help us confirm what's going on and also evaluate your fix PR.

**Comment by katiedeangelis** [comment]
Hi Alice, thanks for reaching out. Unfortunately I ran across this bug will putting a Stencil POC together for use in our atomic design system where I work. We’ve decided not to move forward with Stencil, but thought I’d add and fix the bug while I was working on it. I don’t have time to spin up a new repo, so I understand if you decide not to move forward with the PR. Thanks!

**Comment by rwaskiewicz** [comment]
Relabeling for the team to take a closer look. Thanks!

**Comment by tanner-reits** [comment]
A fix for this was included in today's [v4.19.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.19.0)!
